src/Entity/OgMembership.php

```diff
@@ -126,36 +128,40 @@ public function getFieldName() {
     return $this->get('field_name')->value;
   }
 
+
   /**
    * {@inheritdoc}
    */
-  public function setEntityId($gid) {
-    $this->set('entity_id', $gid);
+  public function setGroup(EntityInterface $group) {
+    $this->set('entity_type', $group->getEntityTypeId());
+    $this->set('entity_id', $group->id());
     return $this;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function getEntityId() {
-    return $this->get('entity_id')->value;
+  public function getGroupEntityType() {
+    return $this->get('entity_type')->value;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function setGroupEntityType($groupType) {
-    $this->set('entity_type', $groupType);
-    return $this;
+  public function getGroupId() {
+    return $this->get('entity_id')->value;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function getGroupEntityType() {
-    return $this->get('entity_type')->value;
+  public function getGroup() {
+    $entity_type = $this->get('entity_type')->value;
+    $entity_id = $this->get('entity_id')->value;
+    return \Drupal::entityTypeManager()->getStorage($entity_type)->load($entity_id);
   }
```

Comment:
Oh wow, great idea! Retrieving the group entity from the membership will come in really handy!